src/main/software/amazon/event/ruler/ByteMachine.java

```diff
@@ -810,9 +809,13 @@ NameState findPattern(final Patterns pattern) {
 
     private NameState findAnythingButPattern(AnythingBut pattern) {
 
-        Set<NameState> nextNameStates = pattern.getValues().stream().
-                map(value -> findMatchPattern(getParser().parse(pattern.type(), value), pattern)).
-                filter(Objects::nonNull).collect(Collectors.toSet());
+        Set<NameState> nextNameStates = new HashSet<>(pattern.getValues().size());
+        for (String value : pattern.getValues()) {
+            NameState matchPattern = findMatchPattern(getParser().parse(pattern.type(), value), pattern);
+            if (matchPattern != null) {
+                nextNameStates.add(matchPattern);
+            }
+        }
```

Comment:
super-duper nitpicking, but I think you can have a consistent style here and in `ByteMap.getTransitions()` if you for `x.forEach( elem -> addWhenNotNull (...) )`. I'm not sure on the performance implications yet but maybe slightly readable. Maybe...

Reply:
I did update this one to the new style that I believe is easier to understand.